internal/query/parse_test.go

```diff
@@ -104,6 +104,21 @@ func TestParseSearchParamsSuccessfully(t *testing.T) {
 	}
 }
 
+func TestNestingInArrayErrorCase(t *testing.T) {
+	u, err := url.Parse("?metadata[key1][][key2]=value1&metadata[key1][][key2]=value2")
+	require.NoError(t, err)
+
+	c := &gin.Context{
+		Request: &http.Request{
+			URL: u,
+		},
+	}
+
+	_, err = ParseSearchParams[ExampleConditionsForTests](c)
+	require.Error(t, err)
```

Comment:
From docs: 
```ErrorContains asserts that a function returned an error (i.e. not `nil`) and that the error contains the specified substring```

So it looks like we can remove this line :)

Reply:
Sure :) 
Changed